File: java/src/test/java/problems1_100/AddTwoNumbersTest.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class AddTwoNumbersTest extends SampledTest {
    @Test
    public void test() {
        for (AddTwoNumbers solution : AddTwoNumbers.values()) {
            for (AddTwoNumbersSample sample : getSamples(AddTwoNumbersSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.addTwoNumbers(
                            SinglyListNode.of(sample.input.l1),
                            SinglyListNode.of(sample.input.l2)
                        )
                    ).isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/AddTwoNumbersTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class AddTwoNumbersTest : SampledTest() {
    @Test
    fun test() =
        AddTwoNumbers.entries.forEach { solution ->
            getSamples<Array<AddTwoNumbersSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(
                        solution.addTwoNumbers(
                            SinglyListNode.of(*input.l1),
                            SinglyListNode.of(*input.l2),
                        ),
                    ).isEqualTo(SinglyListNode.of(*output!!))
            }
        }
}
